### PR TITLE
Port upgrades to lmt apps

### PR DESCRIPTION
**Pull request by chilliwilly77**
* Metrics Controller (as per cya) 

 * Using Controller/Models index.js pattern (as per cya) 

 * Tests: Have Page classes extend a base Page class with useful methods/contrstuctor (as per cya) 

 * Use a routes.js for all test routes (as per drl) 

 * Add tests for the EXPRESS_BASE_PATH being set (as per drl) 

 * Remove feature tests, not used anymore 

 * Fix rev versioning issue for assets in gulp file (as per cya). This might need to be ported to DRL 

 * Update Dockerfile to set PORT to 80 (as per cya)

### CODE REVIEW

File: app/app.js [hunk 1]
```diff
@@ -75,9 +69,14 @@ app.use(assetPath, express.static(path.join(__dirname, '..', 'dist', 'public')))
 
 app.use(helmet.noCache());
 
-app.use(`${basePath}/`, indexController);
-app.use(`${basePath}/`, cookieController);
-app.use(`${basePath}/users`, usersController);
+if (process.env.MI === 'true' || process.env.MI === true) {
```
Comment (mangogrin) [inline app/app.js]: are these conditions different?

File: app/locales/en.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
     }
   },
   "error": {
+    "400": "We're experiencing technical problems.",
```
Comment (mangogrin) [inline app/locales/en.json]: Shouldn't this message reflect that something is wrong with the request itself?

File: app/routes.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class Routes {
+  constructor(basePath) {
+    this.basePath = basePath;
+  }
+
+  baseUrl(url) {
+    return `${this.basePath}${url}`;
+  }
+
+  mainUrl() {
```
Comment (mangogrin) [inline app/routes.js]: probabily this function should be called baseUrl (from the field name)?

File: test/common/page_objects/page.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
+class Page {
```
Comment (mangogrin) [inline test/common/page_objects/page.js]: 👍

File: app/app.js (hunk 1 above)
Comment (chilliwilly77) [reply app/app.js]: yeah in JS they are

**Pull request merged**